elixir/apps/domain/lib/domain/auth/adapters/microsoft_entra/api_client.ex

```diff
@@ -0,0 +1,158 @@
+defmodule Domain.Auth.Adapters.MicrosoftEntra.APIClient do
+  use Supervisor
+
+  @pool_name __MODULE__.Finch
+
+  def start_link(_init_arg) do
+    Supervisor.start_link(__MODULE__, nil, name: __MODULE__)
+  end
+
+  @impl true
+  def init(_init_arg) do
+    children = [
+      {Finch,
+       name: @pool_name,
+       pools: %{
+         default: pool_opts()
+       }}
+    ]
+
+    Supervisor.init(children, strategy: :one_for_one)
+  end
+
+  defp pool_opts do
+    transport_opts =
+      Domain.Config.fetch_env!(:domain, __MODULE__)
+      |> Keyword.fetch!(:finch_transport_opts)
+
+    [conn_opts: [transport_opts: transport_opts]]
+  end
+
+  def list_users(api_token) do
+    endpoint =
+      Domain.Config.fetch_env!(:domain, __MODULE__)
+      |> Keyword.fetch!(:endpoint)
+
+    uri =
+      URI.parse("#{endpoint}/v1.0/users")
+      |> URI.append_query(
+        URI.encode_query(%{
+          "$select" =>
+            Enum.join(
+              ~w[
+                id
+                accountEnabled
+                displayName
+                givenName
+                surname
+                mail
+                userPrincipalName
+              ],
+              ","
+            ),
+          "$filter" => "accountEnabled eq true"
+        })
+      )
+
+    list_all(uri, api_token)
+  end
+
+  def list_groups(api_token) do
+    endpoint =
+      Domain.Config.fetch_env!(:domain, __MODULE__)
+      |> Keyword.fetch!(:endpoint)
+
+    uri =
+      URI.parse("#{endpoint}/v1.0/groups")
+      |> URI.append_query(
+        URI.encode_query(%{
+          "$select" =>
+            Enum.join(
+              ~w[
+                id
+                description
+                displayName
+              ],
+              ","
+            )
+        })
+      )
+
+    list_all(uri, api_token)
+  end
+
+  def list_group_members(api_token, group_id) do
+    endpoint =
+      Domain.Config.fetch_env!(:domain, __MODULE__)
+      |> Keyword.fetch!(:endpoint)
+
+    uri =
+      URI.parse("#{endpoint}/v1.0/groups/#{group_id}/members")
+      |> URI.append_query(
+        URI.encode_query(%{
+          "$select" =>
+            Enum.join(
+              ~w[
+                id
+                accountEnabled
+                description
+                displayName
+              ],
+              ","
+            )
```

Comment:
Possibly.  I thought I had tested users being removed from a group when they were set to `inactive`, but I need to double check.

Reply:
Since this is outdated, it's not show me the whole code snippet, but if this was for the group membership call, I put a comment in the code as to why I didn't include the `$filter` on that call.